contribution.md

```diff
@@ -0,0 +1,125 @@
+script for this purpose. After cloning the repo run following command in your preferred terminal.
+
+```
+sh developing/check.sh
+```
+
+If everything is installed then script should print success messages.
+If all pre-requisite are satisfied, [install](#installing-dependencies) dependencies.
+Follow following steps for installing the missing dependencies.
+
+### Installing Rust
+To install the rust, use the Rust version manager called **rust** which will automatically install
+the Rust on local. Run following command in the terminal.
+
+```
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+
+Once command finishes processing, check the installed Rust version using
+
+```
+rustc --version
+```
+
+which should print the Rust version on terminal.
+
+### Installing NodeJS
+To install the NodeJS, install NVM(Node Version Manager). Follow [installation](https://github.com/nvm-sh/nvm) guide for installing NVM on local.
+Once NVM in ready on your local, install NodeJS using
+
+```
+nvm install --lts
+```
+
+which will install latest long term NodeJS version. Check if node is installed or not using
+
+```
+node -v
+```
+
+which will print the installed NodeJS version on terminal.
+
+### Installing Ruby
+You must have latest Ruby installed on your system. Prefer your choice of version manager
+either [RBENV](https://github.com/rbenv/rbenv) or [RVM](https://rvm.io/).
+Once version manager is ready to use, install Ruby version `>=2.7.7`.
+
+### Installing dependencies
+This project uses few dependencies from other languages and to install them run
+following command in terminal.
+
+```
+sh developing/install.sh
+```
+
+We are using [yarn](https://yarnpkg.com/) as a package manager. Installing yarn is simple
+as you just have to run following command in the terminal.
+
+```
+npm install -g yarn
+```
+
+## Build
+As mentioned earlier backend is built in Rust which takes care of reading the log files
+and frontend app is built using Electron JS.
+
+To run the application, execute the following commands.
+
+```
+cd application/holder
+rake developing:holder_bindings
+yarn start electron
+```
+
+which will open Chipmunk user interface, Yay!
+
+### Changes made in Rust backend
+Everytime you change the Rust code you have to compile the binary and
+then start the application using NodeJS script.
+
+```
+cd application/holder
+rake developing:holder_bindings
+yarn start electron
+```
+
+### Changes in ElectronJS code
+You don't need to update the Rust binary, just run the ElectronJS application.
+
+```
+cd application/holder
+yarn run electron
+```
+
+### Changes in Frontend part
+```
+cd application/holder
+rake build:client_dev
+yarn run electron
+```
+
+### Changes in both backend and frontend
+```
+cd application/holder
+rake developing:holder_platform_bindings
+yarn run electron
+```
+
+With this you can take a look at the open issues and open a PR to fix them.
+
```

Comment:
Probably make sense to add here something like:
### Build/developing commands
Full list of commands you can see with 
```
rake -T
```
Also you can use interactive mode with starting `rake` without any arguments
![Peek 2023-04-10 10-09](https://user-images.githubusercontent.com/12843747/230857595-0d0cb497-4589-4b6e-8352-dec842cb6bcf.gif)